fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/detail/event/eventlist/EventListAdapter.java

```diff
@@ -240,51 +170,47 @@ private Event getEvent(int position) {
    * sure the position is occupied by a header or this will throw an exception
    */
   private EventCategory getHeaderCategory(int position) {
-    int nbrOfFutureEvents = Objects.requireNonNull(eventsMap.get(FUTURE)).size();
     int nbrOfPresentEvents = Objects.requireNonNull(eventsMap.get(PRESENT)).size();
-
     if (position == 0) {
-      return PRESENT;
+      // If this function is called, it means that getSize() > 0. Therefore there are some events
+      // in either past or present (or both). If there are none in the present the first item is the
+      // past header
+      return nbrOfPresentEvents > 0 ? PRESENT : PAST;
     }
     int eventAccumulator = 0;
     if (expanded[PRESENT.ordinal()]) {
       eventAccumulator += nbrOfPresentEvents;
```

Comment:
not sure why "+=" here ?

Reply:
Because if the category is expanded we need to tell the adapter to display those events. Therefore we add the number of present events to the accumulator